File: app/src/main/java/ga/forntoh/bableschool/utils/GsonDeserializeExclusion.java
package ga.forntoh.bableschool.utils;

import com.google.gson.ExclusionStrategy;
import com.google.gson.FieldAttributes;

import java.text.SimpleDateFormat;

public class GsonDeserializeExclusion implements ExclusionStrategy {
    @Override
    public boolean shouldSkipField(FieldAttributes f) {
        return f.getDeclaredClass() == SimpleDateFormat.class;
    }

    @Override
    public boolean shouldSkipClass(Class<?> clazz) {
        return false;
    }
}

File: app/src/main/java/ga/forntoh/bableschool/utils/GsonDeserializeExclusion.kt
package ga.forntoh.bableschool.utils

import com.google.gson.ExclusionStrategy
import com.google.gson.FieldAttributes

import java.text.SimpleDateFormat

class GsonDeserializeExclusion : ExclusionStrategy {
    override fun shouldSkipField(f: FieldAttributes): Boolean = f.declaredClass == SimpleDateFormat::class.java

    override fun shouldSkipClass(clazz: Class<*>): Boolean = false
}
